components/tabs/src/storage.rs

```diff
@@ -226,8 +302,41 @@ impl TabsStorage {
     pub(crate) fn wipe_local_tabs(&self) {
         self.local_tabs.replace(None);
     }
+
+    pub(crate) fn put_meta(&mut self, key: &str, value: &dyn ToSql) -> Result<()> {
+        if let Some(db) = self.open_if_exists()? {
+            db.execute_cached(
+                "REPLACE INTO moz_meta (key, value) VALUES (:key, :value)",
+                &[(":key", &key as &dyn ToSql), (":value", value)],
+            )?;
+        }
+        Ok(())
+    }
+
+    pub(crate) fn get_meta<T: FromSql>(&mut self, key: &str) -> Result<Option<T>> {
+        match self.open_if_exists() {
+            Ok(Some(db)) => {
+                let res = db.try_query_one(
+                    "SELECT value FROM moz_meta WHERE key = :key",
+                    &[(":key", &key)],
+                    true,
+                )?;
+                Ok(res)
+            }
+            Err(e) => Err(e),
+            Ok(None) => Ok(None),
+        }
+    }
+
+    pub(crate) fn delete_meta(&mut self, key: &str) -> Result<()> {
+        if let Some(db) = self.open_if_exists()? {
+            db.execute_cached("DELETE FROM moz_meta WHERE key = :key", &[(":key", &key)])?;
+        }
+        Ok(())
+    }
```

Comment:
We've written these so many times, maybe it would make sense to put the code in `sql-support`.  Probably better as a follow-up though.